transports/http/handlers/webhooks_handler.go

```diff
@@ -0,0 +1,105 @@
+package handler
+
+import (
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+)
+
+// nolint: godot
+// registerWebhook godoc.
+//
+//	@Summary Register new webhook
+//	@Tags webhooks
+//	@Accept json
+//	@Produce json
+//	@Success 200 {object} domains.Webhook
+//	@Router /webhook [post]
+//	@Param data body http.WebhookRequest true "Webhook to register"
+//
+// @Security Bearer
+func (h *Handler) registerWebhook(c *gin.Context) {
+	var reqBody WebhookRequest
+	err := c.Bind(&reqBody)
+
+	if err == nil {
+		if reqBody.Url == "" {
+			c.JSON(http.StatusBadRequest, "Url is required")
+			return
+		}
+
+		webhook, err := h.services.Webhooks.CreateWebhook(reqBody.RequiredAuth.Type, reqBody.RequiredAuth.Header, reqBody.RequiredAuth.Token, reqBody.Url)
+		if err == nil {
+			c.JSON(http.StatusOK, webhook)
+		} else if webhook == nil {
+			c.JSON(http.StatusOK, err.Error())
+		}
+	}
+
+	if err != nil {
+		c.JSON(http.StatusBadRequest, err.Error())
+	}
```

Comment:
IMHO reversing the error checks (I mean first checking if there is an error)
is more Go common and actually looks more clear.

```suggestion
	var reqBody WebhookRequest
	if err := c.Bind(&reqBody); err != nil {
		c.JSON(http.StatusBadRequest, err.Error())
		return
	}


	if reqBody.Url == "" {
		c.JSON(http.StatusBadRequest, "Url is required")
		return
	}

	webhook, err := h.services.Webhooks.CreateWebhook(reqBody.RequiredAuth.Type, reqBody.RequiredAuth.Header, reqBody.RequiredAuth.Token, reqBody.Url)
	if err != nil {
		c.JSON(http.StatusBadRequest, err.Error())
		return
	} 
	
	c.JSON(http.StatusOK, webhook)
```